tests/integration-tests/tests/iam/test_iam_image.py

```diff
@@ -0,0 +1,113 @@
+# Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "LICENSE.txt" file accompanying this file.
+# This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, express or implied.
+# See the License for the specific language governing permissions and limitations under the License.
+import logging
+import time
+
+import boto3
+import pytest
+from assertpy import assert_that
+from utils import generate_stack_name
+
+from tests.common.utils import retrieve_latest_ami
+
+
+@pytest.mark.usefixtures("os", "instance")
+def test_iam_roles(
+    request,
+    region,
+    os,
+    create_roles_stack,
+    pcluster_config_reader,
+    images_factory,
+    test_datadir,
+):
+    # Create build image roles
+    image_roles_stack = create_roles_stack(
+        stack_prefix="integ-tests-iam-image-roles", roles_file="image-roles.cfn.yaml"
+    )
+    lambda_cleanup_role = image_roles_stack.cfn_outputs["BuildImageLambdaCleanupRole"]
+    instance_profile = image_roles_stack.cfn_outputs["BuildImageInstanceProfile"]
+    # instance_role = image_roles_stack.cfn_outputs["BuildImageInstanceRole"]
+
+    # Generate image ID
+    image_id = generate_stack_name("integ-tests-build-image", "")
+
+    # Get base AMI
+    base_ami = retrieve_latest_ami(region, os, ami_type="pcluster", architecture="x86_64")
+
+    image_config = pcluster_config_reader(
+        config_file="image.config.yaml",
+        parent_image=base_ami,
+        instance_profile=instance_profile,
+        lambda_cleanup_role=lambda_cleanup_role,
+    )
+
+    image = images_factory(image_id, image_config, region)
+    pcluster_describe_image_result = image.describe()
+    logging.info(pcluster_describe_image_result)
+
+    cfn_client = boto3.client("cloudformation", region_name=region)
+    ec2_client = boto3.client("ec2", region_name=region)
+    lambda_client = boto3.client("lambda", region_name=region)
+
+    roles_checked = 0
+    while image.image_status == "BUILD_IN_PROGRESS":
+        time.sleep(600)
+        pcluster_describe_image_result = image.describe()
+        logging.info(pcluster_describe_image_result)
```

Comment:
it would be nice to have this waiter as a function, optionally you could use the retry annotation. Also I'd check every minute rather than every 10 otherwise in the worst case scenario where the image completes right after we checked we spend 10 minutes waiting.

Reply:
done